vault/logical_system_custom_messages.go

```diff
@@ -577,20 +587,31 @@ func (b *SystemBackend) handleUpdateCustomMessage(ctx context.Context, req *logi
 	}
 
 	var link *uicustommessages.MessageLink
-	if linkMap != nil {
-		link = &uicustommessages.MessageLink{}
-
-		for k, v := range linkMap {
-			href, ok := v.(string)
-			if !ok {
-				return logical.ErrorResponse("invalid url for %q key link parameter value", k), nil
-			}
-
-			link.Title = k
-			link.Href = href
+FORLOOP:
+	for k, v := range linkMap {
```

Comment:
Since we're doing the same validation in two places, it might make sense to refactor this out into a helper method

Reply:
Given the refactoring you suggested, the extraction to a function should be a lot easier.